org.civicrm.reqangular/src/common/services/format-currency.service.js

```diff
@@ -2,85 +2,123 @@
 
 define([
   'common/modules/services',
-  'common/services/api'
+  'common/services/api',
+  'common/models/settings.model'
 ], function (services) {
   'use strict';
 
   services.factory('FormatCurrencyService', FormatCurrencyService);
 
-  FormatCurrencyService.$inject = ['api', '$q', '$log'];
+  FormatCurrencyService.$inject = ['$log', '$q', 'Settings'];
 
-  function FormatCurrencyService (api, $q, $log) {
+  function FormatCurrencyService ($log, $q, Settings) {
     $log.debug('Service: FormatCurrencyService');
 
-    var promise;
-
     return {
-      addSeperators: addSeperators,
-      removeCharacters: removeCharacters
+      format: format
     };
 
     /**
-     * Adds defined thousand and decimal separators in the given amount
-     *
-     * @param {string} amount
-     * @return {string}
-     */
-    function addSeperators (amount) {
-      return fetchSeparators().then(function (separators) {
-        // Add thousand seperator to amount using regex expression
-        var withThousandSeparator = amount.toString().replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1' + separators.thousand);
-        var formattedCurrency = withThousandSeparator.replace('.', separators.decimal);
-
-        return (+amount) ? formattedCurrency : ('0' + separators.decimal + '00');
+    * Adds defined thousand and decimal separators in the given amount
+    *
+    * @param {string} amount
+    * @return {promise}
+    */
+    function format (amount) {
+      return Settings.fetchSeparators().then(function (separators) {
+        return {
+          formatted: addSeparators(amount, separators),
+          unformatted: removeSeparators(amount, separators)
```

Comment:
I would use the word `original` instead of unformatted since unformatted might mean "formatted and then reverted back"

Reply:
yeah, something like that

Reply:
@igorpavlov Actually, the resulted value for `unformatted` is a value with decimal separator as "." With sole purpose of making the formatted value to be used in calculations. So, i think calling it `original` is not suitable either. I prefer to call `calculable` or `asNumber`. What do you say?

cc @AkA84

Reply:
@bmnepali  `parsed` ?

Reply:
`parsed` as well looks fine.